crates/aggchain-proof-contracts/src/tests/mod.rs

```diff
@@ -0,0 +1,530 @@
+            .match_body(mockito::Matcher::Json(get_local_exit_root_body))
+            .with_body(
+                json!({
+                    "id": 1,
+                    "jsonrpc": "2.0",
+                    "result": "0x27ae5ba08d7291c96c8cbddcc148bf48a6d68c7974b94356f53754ef6171d757"
+                })
+                .to_string(),
+            )
+            .create();
+
+        let result = contracts_client.get_l2_local_exit_root(10).await;
+
+        mock_l2.assert_async().await;
+        let local_exit_root = result?;
+        assert_eq!(
+            local_exit_root,
+            aggchain_proof_core::Digest(
+                B256::from_str(
+                    "0x27ae5ba08d7291c96c8cbddcc148bf48a6d68c7974b94356f53754ef6171d757"
+                )
+                .unwrap()
+                .0
+            )
+        );
+
+        Ok(())
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_invalid_local_exit_root() -> Result<(), Box<dyn std::error::Error>> {
+        let (contracts_client, test_servers) = aggchain_contracts_rpc_client().await?;
+        let mut server_l2_el = test_servers.server_l2_el;
+
+        // We ask the PolygonZkEVMBridgeV2 for the local exit root with `getRoot()`
+        let get_local_exit_root_body = serde_json::json!({
+            "method": "eth_call",
+            "params": [{
+                "to":"0xd81e7fb88b8e3a6bae6c1b64e11f1a355641fb7c",
+                "input":"0x5ca1e165",
+            },
+            "0xa"],
+            "id": 1,
+            "jsonrpc": "2.0",
+        });
+
+        let mock_l2 = server_l2_el
+            .mock("POST", "/")
+            .with_status(200)
+            .with_header("content-type", "text/javascript")
+            .match_body(mockito::Matcher::Json(get_local_exit_root_body))
+            .with_body(
+                json!({
+                    "id": 1,
+                    "jsonrpc": "2.0",
+                    "result": "0x27ae5ba08d7291c96c8cbddcc"
+                })
+                .to_string(),
+            )
+            .create();
+
+        let result = contracts_client.get_l2_local_exit_root(10).await;
+
+        mock_l2.assert_async().await;
+        match result {
+            Err(crate::Error::LocalExitRootError(alloy::contract::Error::TransportError(_))) => {
+                Ok(())
+            }
+            Err(crate::Error::LocalExitRootError(error)) => {
+                panic!(
+                    "Expected alloy transport deserialization error, got {:?}",
+                    error
+                );
+            }
+            Err(e) => panic!("Expected LocalExitRootError, got {:?}", e),
+            Ok(_) => panic!("Expected LocalExitRootError, got valid Digest"),
+        }
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn get_rollup_config_hash() -> Result<(), Box<dyn std::error::Error>> {
+        let (contracts_client, test_servers) = aggchain_contracts_rpc_client().await?;
```

Comment:
General question: Do you have any guidance as to when to use `?` vs `.unwrap()` in tests? As far as I can tell the effect is more or less the same. This test uses both.

Reply:
Yeah, I should be consistent with one. I prefer errors, give more info why the test failed.